faculty_cli/cli.py

```diff
@@ -1451,3 +1450,147 @@ def dataset_ls(project, prefix, show_hidden):
+@click.option(
+    "--version",
+    default=None,
+    help="The specific version of the template to use.",
+)
+@click.option(
+    "--directory",
+    default=Path.cwd(),
+    help="The optional target directory where the template will be placed.",
+    type=click.Path(),
+)
+def clone(source_template_name, version, directory):
+    """Clone an existing template."""
+    print(source_template_name)
+    print(directory)
+
+    if version is None:
+        # TODO: Get the latest version
+        print(version)
+
+
+@template.command()
+@click.argument("project_identifier")
+@click.option(
+    "--source_directory",
+    default=Path.cwd(),
+    type=click.Path(),
+    help="Path to the source template directory where the template"
+    + "will be placed.",
+)
+@click.option(
+    "--target-directory",
+    default=Path("/"),
+    help="The specific version of the template to use.",
+)
+@click.option(
+    "parameters",
+    "--parameter",
+    "-p",
+    type=(str, str),
+    default=[],
+    help="The optional target directory where the template will be placed.",
+    multiple=True,
+)
+def apply_from_directory(
+    project_identifier, source_directory, target_directory, parameters
+):
+    """Apply from a directory to a project."""
+    project_id = _resolve_project(project_identifier)
+    print(project_id)
+    print(source_directory)
+    print(target_directory)
+
+    for (key, value) in parameters:
+        print(key, value)
+
+
+@template.command()
+@click.argument("project_name")
+@click.option(
+    "--source_directory",
+    default=Path.cwd(),
+    type=click.Path(),
+    help="Path to the source template directory where the template."
+    + "will be placed.",
+)
+@click.option(
+    "--target-directory",
+    default=Path("/"),
+    help="The specific version of the template to use.",
+)
+@click.option(
+    "parameters",
+    "--parameter",
+    "-p",
+    type=(str, str),
+    default=[],
+    help="The optional target directory where the template will be placed.",
+    multiple=True,
+)
+def create_project_from_directory(
+    project_name, source_directory, target_directory, parameters
+):
+    """Apply from a directory to a new project."""
+    print(project_name)
+    print(source_directory)
+    print(target_directory)
+
+    for (key, value) in parameters:
+        print(key, value)
+
+
+@template.group()
+def publish():
+    """Publish to the knowledge centre."""
+    pass
+
+
+@publish.command(name="new")
+@click.argument("template_name")
+@click.option(
+    "--source_directory",
+    default=Path.cwd(),
+    type=click.Path(),
+    help="Path to the source template directory where the"
+    + "template will be placed.",
+)
+def publish_new_template(template_name, source_directory):
+    """Publish a new template from a directory to the knowledge centre."""
+    print(template_name)
+    print(source_directory)
+
+
+@publish.command(name="version")
+@click.argument("template_name")
```

Comment:
For consistency with other CLI commands:
```suggestion
@click.argument("template")
```

Reply:
I like `template_name` here to distinguish between the `clone` command (and potentially others we could imagine adding) that take `template` with `template` being a UUID or the template_name. 

What do you think?

Reply:
I think that people almost never pass the UUIDs on the command line and so the semantics of the CLI should by default assume that people are always passing names. I think that shorter, pithier names also make for a more consumable usage string in the generated help, e.g.:

```
Usage: faculty template publish new TEMPLATE [SOURCE]
```

rather than

```
Usage: faculty template publish new TEMPLATE_NAME [SOURCE_DIRECTORY]
```

I don't think users will really appreciate the difference between TEMPLATE and TEMPLATE_NAME in different subcommands and so it might be a source of confusion for simple use cases.

Any thoughts on this?

Reply:
That's fair. I'll make that change